docs/basics/overview.md

```diff
@@ -0,0 +1,232 @@
+- **Note**: As of the current version, flow matching models tend to be more stable and easier to train than diffusion models. This may change in future releases.
+
+**Flow Matching is the recommended default in GenSBI.**
+
+## How Components Work Together
+
+Here's what happens during training:
+
+1. **Data Loading**: The pipeline gets batches of (observations, conditions) from your dataset.
+
+2. **Loss Computation**:
+   - Sample random time steps `t ∈ [0, 1]`
+   - Create noisy versions of the data based on `t`
+   - The model (unwrapped) predicts the velocity/noise as a function of (obs, cond, t)
+   - Compare prediction to ground truth
+
+3. **Optimization**:
+   - Compute gradients
+   - Update model parameters
+   - Update EMA shadow weights
+
+4. **Validation**:
+   - Periodically evaluate on validation set
+   - Save checkpoints if performance improves
+   - Early stopping if validation loss diverges
+
+During inference:
+
+1. **ODE Solving** (Flow Matching):
+   - Wrap the model to provide standard interface for the solver
+   - Start with Gaussian noise
+   - Use the wrapped model's `get_vector_field()` method with an ODE solver
+   - Result: samples from the posterior distribution
+
+2. **Iterative Denoising** (Diffusion):
+   - Wrap the model for the SDE sampler
+   - Start with pure noise
+   - Iteratively denoise using the learned denoiser
+   - Result: samples from the posterior distribution
+
+## File Organization
+
+The codebase is organized into logical modules:
+
+```
+src/gensbi/
+├── models/              # Neural network architectures
+│   ├── flux1/          # Flux1 model
+│   ├── flux1joint/     # Flux1Joint model
+│   ├── simformer/      # Simformer model
+│   ├── wrappers/       # Time/noise handling wrappers
+│   └── losses/         # Loss functions
+├── recipes/             # High-level training pipelines
+│   ├── flux1.py
+│   ├── simformer.py
+│   └── ...
+├── flow_matching/       # Flow matching components
+│   ├── path/           # Interpolation paths
+│   ├── solver/         # ODE solvers
+│   └── loss/           # Flow matching loss
+├── diffusion/           # Diffusion components
+│   ├── sampler/        # Diffusion samplers
+│   ├── sde/            # SDE definitions
+│   └── loss/           # Diffusion loss
+└── utils/               # Utility functions
+```
+
+## Design Principles
+
+GenSBI follows these design principles:
+
+1. **Modularity**: Components (models, wrappers, losses, solvers) are independent and composable.
+
+2. **Sensible Defaults**: Pipelines come with reasonable default hyperparameters that work for many problems.
+
+3. **Easy Customization**: You can override specific methods (e.g., optimizer, loss function) without rewriting everything.
+
+4. **JAX-Native**: Built on JAX and Flax NNX for performance, automatic differentiation, and hardware acceleration.
+
+5. **SBI-Focused**: Designed specifically for simulation-based inference, not general-purpose generative modeling.
```

Comment:
I don't like this phrase. gensbi is meant for conditional and unconditional density estimation, and it has applications for simulation based inference, but more in general for any kind of conditional density estimation task, such as neural posterior density estimation, neural likelihood estimation, and neural prior estimation (that is, learn to sample from a distribution of data, without conditioning on anything), so ideally anything that involves learning a distribution from the data